### PR TITLE
Enumerate all roles / permissions for Pubsub IAM.

### PR DESCRIPTION
**Pull request by tseaver**
Capture pubsub-specific 'publisher' and 'subscriber' roles.

See: https://cloud.google.com/pubsub/access_control#permissions

#1687 points out that `pubsub.topics.testIamPermissions` and `pubsub.subscriptions.testIamPermissions` take lists of *permissions*, rather than the roles which imply them.

### CODE REVIEW

**Comment by dhermes** [comment]
Whoa!

@tmatsuo Does this look OK to you?

**Comment by rimey** [comment]
@tseaver Could you summarize the reasons for modeling this IAM structure in the client library?

**Comment by tseaver** [comment]
We previously exposed the three "generic" IAM roles as constants (I originally had them private, but made them public at @dhermes' urging).  This PR adds the roles which are specific to Pubsub, and also enumerates the permissions as constants (needed by those who would call `Topic.check_iam_permissions` or `Subscription.check_iam_permissiosns`).  One reason for the constants is that the back-end raises 400 errors if one passes an unknown permission string to `testIamPermissions` (see #1687).

**Comment by rimey** [comment]
@tseaver Thank you for that explanation.

**Comment by tmatsuo** [comment]
@dhermes Yeah it makes sense to me

**Comment by dhermes** [comment]
OK @tseaver LGTM to merge based on Takashi's sign off
